src/avram/associations/has_many.cr

```diff
@@ -180,16 +181,16 @@ module Avram::Associations::HasMany
     end
 
     private def lazy_load_{{ assoc_name }} : Array({{ model }})
+      {% query = query || "#{model}::BaseQuery.new".id %}
       {% if through %}
-        through_results = {{ through.first.id }}_query.preload_{{ through[1].id }}.results
+        through_results = {{ through.first.id }}_query.preload_{{ through[1].id }}({{ query }}).results
```

Comment:
In my spec, this would expand to `taggings_query.preload_tags(Tag::BaseQuery.new.name(...).results)`. But I'm not sure if we can always just pass in the BaseQuery like that and have it work. Maybe this line has to change depending on if a `query` is passed in?

Reply:
see my comment above to default it if it's nil